### PR TITLE
Add unitests for  detr 3.x

### PR DESCRIPTION
**Pull request by LYMDLUT**
We (me, @jshilong, @Li-Qingyun , and @KeiChiTse) are going to refactor DETR-like models, to enhance the usability and readability of our codebase.
This pr aims to refine code for ci test.

### CODE REVIEW

**Comment by Li-Qingyun** [comment]
Don't forget to check unit test~

File: tests/test_models/test_layers/test_transformer.py [hunk 1]
```diff
@@ -529,5 +529,3 @@ def test_detr_transformer_dencoder_encoder_layer():
     with pytest.raises(AssertionError):
         # len(operation_order) == 6
         DetrTransformerEncoder(**config)
-
```
Comment (Li-Qingyun) [inline tests/test_models/test_layers/test_transformer.py:532]: Leave a blank line.

File: mmdet/models/detectors/deformable_detr.py [hunk 2]
```diff
@@ -539,3 +539,29 @@ def get_proposal_pos_embed(proposals: Tensor,
...
+        Returns:
+            tuple[Tensor]: A tuple of features from ``bbox_head`` forward.
+        """
+        img_feats = self.extract_feat(batch_inputs)
+        head_inputs_dict = self.forward_transformer(img_feats,
+                                                    batch_data_samples)
+        head_inputs_dict.pop('enc_outputs_class')
+        head_inputs_dict.pop('enc_outputs_coord')
+        results = self.bbox_head.forward(**head_inputs_dict)
+        return results
```
Comment (Li-Qingyun) [inline mmdet/models/detectors/deformable_detr.py:567]: You do not need to write this
There was process here in `self.pre_decoder`.
It is enough to complete this part of the logic.

```Python
def pre_decoder(self, memory: Tensor, memory_mask: Tensor,
                spatial_shapes: Tensor) -> Tuple[Dict, Dict]:
    ......
    head_inputs_dict = dict(
        enc_outputs_class=enc_outputs_class if self.as_two_stage else None,
        enc_outputs_coord=enc_outputs_coord_unact.sigmoid()
        if self.as_two_stage else None)
    return decoder_inputs_dict, head_inputs_dict
```

Comment (LYMDLUT) [reply mmdet/models/detectors/deformable_detr.py:567]: maybe you should try to run it,as _forward does not have enclass and encoord

File: tests/test_models/test_detectors/test_deformable_detr.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+from mmengine.structures import InstanceData
+
+from mmdet.models.detectors.deformable_detr import DeformableDETR
+from mmdet.structures import DetDataSample
+from mmdet.utils import register_all_modules
+
+
+class TestDeformableDETR(TestCase):
+
+    def setUp(self):
```
Comment (jshilong) [inline tests/test_models/test_detectors/test_deformable_detr.py:15]: always use low case  letters for the function name

File: tests/test_models/test_detectors/test_deformable_detr.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            'scale_factor': (1, 1),
+            'pad_shape': (s, s),
+            'batch_input_shape': (s, s)
+        }
+        img_metas = DetDataSample()
+        img_metas.set_metainfo(metainfo)
+        batch_data_samples = []
+        batch_data_samples.append(img_metas)
+
+        config = ConfigDict(
```
Comment (jshilong) [inline tests/test_models/test_detectors/test_deformable_detr.py:32]: you can just load the corresponding config by the relative path

File: tests/test_models/test_detectors/test_detr.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+            'scale_factor': (1, 1),
+            'pad_shape': (s, s),
+            'batch_input_shape': (s, s)
+        }
+        img_metas = DetDataSample()
+        img_metas.set_metainfo(metainfo)
+        batch_data_samples = []
+        batch_data_samples.append(img_metas)
+
+        config = ConfigDict(
```
Comment (jshilong) [inline tests/test_models/test_detectors/test_detr.py:32]: same comment

File: tests/test_models/test_detectors/test_deformable_detr.py (hunk 3 above)
Comment (LYMDLUT) [reply tests/test_models/test_detectors/test_deformable_detr.py:15]: We use function name “setUp” the same as  “https://github.com/Li-Qingyun/mmdetection/blob/3c016c823a5ba4eebb9a0a318461104b0f669732/tests/test_models/test_detectors/test_single_stage.py#L17”。
class TestSingleStageDetector(TestCase):

    def setUp(self):
        register_all_modules()

File: tests/test_models/test_detectors/test_deformable_detr.py (hunk 4 above)
Comment (LYMDLUT) [reply tests/test_models/test_detectors/test_deformable_detr.py:32]: Ok，I will try

File: tests/test_models/test_detectors/test_deformable_detr.py (hunk 3 above)
Comment (jshilong) [reply tests/test_models/test_detectors/test_deformable_detr.py:15]: just fix them togather

**Review by jshilong — APPROVED** [review]
LGTM

**Review by Li-Qingyun — APPROVED** [review]
LGTM

File: tests/test_models/test_detectors/test_deformable_detr.py (hunk 3 above)
Comment (LYMDLUT) [reply tests/test_models/test_detectors/test_deformable_detr.py:15]: 9575e6b1e129b791b0963b49b5e30c5
It is the style of unittest lib,as the function setUp here just overwrote the function setUp in ubittest lib,I think maybe we should not change it.

**Pull request merged**